Vivo Fit 3 support

TaaviE opened the pull request:
As the issues were closed on your repository I made this PR.

If and what are your plans on polishing up and upstreaming the Vivo Fit 3 code to Gadgetbridge, @mjsir911?

mjsir911 committed:
Compiling barebones skeleton

mjsir911 committed:
Checkpoint

mjsir911 committed:
Checkpoint

mjsir911 committed:
Redo upload/download streams & move out io streams

mjsir911 committed:
Handle resource allocation a bit better

mjsir911 committed:
Implement LengthPrefixer

mjsir911 committed:
Add explicit starting value for lengthPrefixer

mjsir911 committed:
Implement CRCAdder

mjsir911 committed:
Noticable value for dummy implementation

mjsir911 committed:
Implement COBSEncoder

mjsir911 committed:
Properly close upload stream on send

mjsir911 committed:
Flush shenanigans

mjsir911 committed:
Successful write & fixes along the way

mjsir911 committed:
Use better deque

mjsir911 committed:
Change onCharacteristicChanged to not change

mjsir911 committed:
Save progress

mjsir911 committed:
Finish up queued actions before trashing queue

mjsir911 committed:
Gracefully dispose

mjsir911 committed:
Don't lock up if queue is empty

mjsir911 committed:
Add SystemEventOperation

mjsir911 committed:
Let the caller deal with transactions

This breaks the operation contract but honestly I'm too tired to care

mjsir911 committed:
Get a slightly more sane api all thanks to @mormegil-cz

Also finally get the proper time set! I don't actually know what the
problem was, maybe a mis-cast or broken order of operations? whoops.

I really like the idea of using a map for this, I didn't think about it
too hard and just had a list of k-v pairs. I wish I could automagically
deal with all possible data types though

mjsir911 committed:
Just use methods instead of indexing all the time

mjsir911 committed:
Add working crc16 class inspired by @mormegil-cz

mjsir911 committed:
send() -> flush()

mjsir911 committed:
Have bytebuffer optionally accept a backing buffer

mjsir911 committed:
Have support handle the outputstream

mjsir911 committed:
Give the CRCAdder an objectoutput to define endianness

mjsir911 committed:
Proper attribution

mjsir911 committed:
Perform responses immediately

mjsir911 committed:
Simplify dispatching workflow

mjsir911 committed:
Clean up

mjsir911 committed:
Implement writeBytes()

mjsir911 committed:
Tidy up SystemEventOperation

mjsir911 committed:
Simplify COBSEncoder to use superclass & no flush

mjsir911 committed:
Add length tracker

mjsir911 commented:
Hi @TaaviE! Thanks for your interest in this. 

For reference, there is some other garmin watch work at [mormegil-cz/Gadgetbridge](https://github.com/mormegil-cz/Gadgetbridge/tree/garmin-wip)



> If and what are your plans on polishing up and upstreaming the Vivo Fit 3 code to Gadgetbridge, @mjsir911?

I would love for the work I've done to be upstreamed but I don't have much experience with contributing changes this large.

Additionally, there is still a lot of work to be done implementing all the features of the vivofit watch. Currently I just about have setting the time working, and all the groundwork for everything else.

Let me know if you have any more questions, or if you are interested in helping out with the gadgetbridge vivofit stuff, would greatly be appreciated :)

TaaviE commented:
> but I don't have much experience with contributing changes this large.

A good start would be to get rid of excess logging, linter warnings and to extract string resources (for localization). After that is done, you can make a PR and you'll most likely be given feedback what to improve and how.

mjsir911 committed:
Fix assertion linter warning, manual assertions

mjsir911 committed:
Fix empty messages

mjsir911 committed:
Kick of time set once device is initialized

di72nn committed:
Refactoring: localize notification handling in GB

* Handle all notifications via GB.
* Create all notification channels in one place.
* Lazily init notification channels.
* Re-create notification channels to update their localized names.

di72nn committed:
Rename notification channels

di72nn committed:
Add a separate "low battery" notification channel

di72nn committed:
Hide timestamp from device notifications

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (1120 of 1120 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1120 of 1120 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1120 of 1120 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.7% (1117 of 1120 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (1120 of 1120 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/es/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1125 of 1125 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1130 of 1130 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 93.3% (1055 of 1130 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nb_NO/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 99.9% (1129 of 1130 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1130 of 1130 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1130 of 1130 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.6% (1126 of 1130 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1131 of 1131 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

vincegre committed:
Translated using Weblate (French)

Currently translated at 98.9% (1119 of 1131 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1131 of 1131 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.6% (1127 of 1131 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1131 of 1131 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1131 of 1131 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

ashimokawa committed:
bump version, update changelog

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1132 of 1132 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1132 of 1132 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1132 of 1132 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 99.9% (1131 of 1132 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

Commit:
Translated using Weblate (Czech)

Currently translated at 100.0% (1132 of 1132 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/cs/

ashimokawa committed:
update xml and fastlane changelogs

TaaviEo committed:
Improved logging in bonding and tried to recover from weirdness

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1134 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (1134 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 93.1% (1056 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nb_NO/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1134 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1134 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1134 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Fossil Hybrid HR: Block dangerous intents by default and move some settings to developer settings submenu

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.5% (1129 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

Commit:
Revert translation id renames

Commit:
Translated using Weblate (Indonesian)

Currently translated at 13.2% (150 of 1134 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/id/

vanous committed:
reduce margin of icons in device_item to keep one row for typical amount
of icons

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (1140 of 1140 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1140 of 1140 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1140 of 1140 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1140 of 1140 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

ashimokawa committed:
Never show GPS or HW versions, just show FW2

The problem is that a lot of devices showed HR which had GPS, which confused people

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1140 of 1140 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

IngrownMink4 committed:
Translated using Weblate (Spanish)

Currently translated at 99.4% (1134 of 1140 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/es/

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1140 of 1140 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

ashimokawa committed:
Support timezones with half an our offset (like UTC+5:30)

Should help #2283

TaaviEo committed:
Added a R8 rule to make sure data classes for Gson don't get optimized out

geekosaurusR3x committed:
Initial support for Amazfit T-Rex Pro

vincegre committed:
Translated using Weblate (French)

Currently translated at 99.9% (1140 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

Commit:
Translated using Weblate (German)

Currently translated at 99.9% (1140 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

ashimokawa committed:
bump version, draft changelog

comradekingu committed:
Translated using Weblate (English)

Currently translated at 99.9% (1140 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/en/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 93.0% (1062 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nb_NO/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1141 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

Commit:
Amazfit T-Rex Pro: Fix for watch face upload and device settings (#2321)

Reviewed-on: https://codeberg.org/Freeyourgadget/Gadgetbridge/pulls/2321
Co-authored-by: GeekosaurusR3x <geekosaurusr3x@noreply.codeberg.org>
Co-committed-by: GeekosaurusR3x <geekosaurusr3x@noreply.codeberg.org>

ashimokawa committed:
Amazit T-Rex Pro: minor cleanups, support flashig AGPS UIHH (assumed working for such a new device)

ashimokawa committed:
update changelogs

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1141 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1141 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (1141 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.0% (1130 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1141 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.4% (1135 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

xmgz committed:
Translated using Weblate (Galician)

Currently translated at 41.8% (478 of 1141 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/gl/

ashimokawa committed:
Amazfit T-Rex Pro: Add barometer to menu items

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 92.9% (1062 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nb_NO/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.4% (1136 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

TaaviEo committed:
Fixed a misspell and declared the conversion function as static

Commit:
Fossil Hybrid HR: Add watchface designer

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Fossil Hybrid HR: Add file extensions to new assets

Commit:
Fossil Hybrid HR: Edit existing watchfaces from app manager cache

Commit:
Fossil Hybrid HR: Use text inputs for widget coordinates

Commit:
Translated using Weblate (Czech)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/cs/

ashimokawa committed:
Pebble: Fix inverted check for firmware version

Regression from 0.57.1.

This could help #2329

vanous committed:
Debug: add testing device manually

vanous committed:
Check ActivityListing charts for null condition, should fix #2322

TaaviEo committed:
Set the class loader to avoid unmarshalling errors

Commit:
Future-proofed Bluetooth permission request and improved broadcast receiver (#2331)

I saw a log line  posted [here](https://codeberg.org/Freeyourgadget/Gadgetbridge/issues/2118) that indicates that the permission requirement might be incorrect in the broadcast receiver. Removed it.

I also future-proofed the BLUETOOTH/BLUETOOTH_ADMIN permissions according to Android 12 Beta documentation, there's probably much more than this that has to be updated to be fully Android 12 compliant though.

Co-authored-by: TaaviE <taavi.eomae@gmail.com>
Reviewed-on: https://codeberg.org/Freeyourgadget/Gadgetbridge/pulls/2331
Co-authored-by: TaaviE <taavie@noreply.codeberg.org>
Co-committed-by: TaaviE <taavie@noreply.codeberg.org>

vanous committed:
add transliteration to PineTime, solves #2334

vpuhoff committed:
Translated using Weblate (Russian)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/ru/

er2off committed:
Translated using Weblate (Russian)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/ru/

vanous committed:
Remove battery data when device is being removed

ashimokawa committed:
bump version, add changelogs

vanous committed:
Add black background to Dark Theme

vanous committed:
add transliteration to BangleJS and WaspOS

vanous committed:
add missing icons to settings items

ikajdan committed:
Translated using Weblate (Polish)

Currently translated at 100.0% (1142 of 1142 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

vanous committed:
remove @layout/preference_checkbox from preferences as it moves icons slightly to the left

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (1143 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1143 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1143 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 92.9% (1062 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nb_NO/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 100.0% (1143 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1143 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Czech)

Currently translated at 100.0% (1143 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/cs/

vanous committed:
Allows to edit the GPX track entry in Sports Activity item

ashimokawa committed:
InfiniTime: support notification for battery level

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1143 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

vanous committed:
add GPX receiver via intent

Commit:
Fossil Hybrid HR: Extend AbstractGBActivity for dark theme support

Commit:
Fossil Hybrid HR: Move preview&save buttons to action bar

Commit:
Fossil Hybrid HR: Show widget previews in watchface designer

Commit:
Fossil Hybrid HR: Add widget position presets and move widget dialog to layout XML

Commit:
Translated using Weblate (Persian)

Currently translated at 11.8% (135 of 1143 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fa/

ashimokawa committed:
Revert "Future-proofed Bluetooth permission request and improved broadcast receiver (#2331)"

This reverts commit e510205b61346a0cfc2375fca1fdfc3c3002f6ee.

Fixes #2342
Fixes #2346

ashimokawa committed:
changelogs, bump version

ashimokawa committed:
tweak changelog

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 100.0% (1150 of 1150 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

Commit:
Translated using Weblate (Czech)

Currently translated at 100.0% (1150 of 1150 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/cs/

Commit:
Fossil Hybrid HR: Replace hardcoded widget type radio buttons with generated dropdown

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 99.9% (1149 of 1150 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1150 of 1150 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1150 of 1150 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

Commit:
Fossil Hybrid HR: Add steps and HR widgets

Commit:
Fossil Hybrid HR: Add drag&drop for widgets

Commit:
Fossil Hybrid HR: Make hardcoded strings translatable

Commit:
Fossil Hybrid HR: Make hardcoded strings translatable

ikajdan committed:
Translated using Weblate (Polish)

Currently translated at 100.0% (1150 of 1150 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

Commit:
Fossil Hybrid HR: Support widget color inversion

pawel-smth committed:
Translated using Weblate (Ukrainian)

Currently translated at 100.0% (1150 of 1150 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/uk/

Commit:
Fossil Hybrid HR: Support configuring watchface internal values

vanous committed:
use '_' instead ':' in exported and imported file names

Commit:
Fossil Hybrid HR: Drop widget on recycle bin to delete

Commit:
Fossil Hybrid HR: Prevent display rotation in watchface designer

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1186 of 1186 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

Commit:
Translated using Weblate (Czech)

Currently translated at 100.0% (1186 of 1186 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/cs/

vanous committed:
fix: apply theme as set by the system

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1186 of 1186 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 99.9% (1185 of 1186 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1186 of 1186 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Added support for SMA Q2 (#2215)

Hi,

i applied the changes from [Emeryths fork](https://github.com/Emeryth/Gadgetbridge) to master in order to get the [SMA Q2 smartwatch](https://hackaday.io/project/85463-color-open-source-smartwatch) support upstream.

It uses [his firmware](https://github.com/Emeryth/sma-q2-oss) (pretty/most popular for this device).

Greatly appreciate any feedback as this is my first attempt to add a device to GB :)

I read through the [tutorial](https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/New-Device-Tutorial), are there other sources for adding new device support?

thanks!

Co-authored-by: x29a <0.x29a.0@gmail.com>
Reviewed-on: https://codeberg.org/Freeyourgadget/Gadgetbridge/pulls/2215
Co-authored-by: x29a <x29a@noreply.codeberg.org>
Co-committed-by: x29a <x29a@noreply.codeberg.org>

ashimokawa committed:
Fix missing breaks in switch, should fix UM25 support

ashimokawa committed:
bump version, draft changelog

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 99.9% (1186 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1187 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

Commit:
Fossil Hybrid HR: Update widget colors and button text

Commit:
Fossil Hybrid HR: Add default Gadgetbridge-themed watchface background

vanous committed:
add System.exit(0) to quit() to ensure exit

IhorHordiichuk committed:
Translated using Weblate (Ukrainian)

Currently translated at 100.0% (1187 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/uk/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1187 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1187 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1187 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 97.8% (1162 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

Commit:
Fossil Hybrid HR: Confirm overwriting cached watchface

Commit:
Translated using Weblate (Russian)

Currently translated at 100.0% (1187 of 1187 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/ru/

Commit:
Fossil Hybrid HR: Rename widget color definitions

Commit:
Fossil Hybrid HR: Add icon for inverting colors of background image

Commit:
Fossil Hybrid HR: Show progress bar while installing watchface

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1189 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

ashimokawa committed:
update changelogs

ashimokawa committed:
fix xml changelog

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1189 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

IhorHordiichuk committed:
Translated using Weblate (Ukrainian)

Currently translated at 100.0% (1189 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/uk/

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1189 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 99.9% (1188 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 92.4% (1099 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nb_NO/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1189 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Fossil Hybrid HR: Add battery and calories widgets

ashimokawa committed:
add fossil-hr-watchface submodule

ashimokawa committed:
add jerryscript as a submodule

ashimokawa committed:
add script to build fossil_hr_watchface by building submodules

ashimokawa committed:
move submodules and build_fossil_hr_watchface.sh to subdir

Commit:
Fossil Hybrid HR: Add 2nd TZ, Active Minutes and Chance Of Rain widgets

ashimokawa committed:
bump gradle and gradle plugin as auto-applied by Android Studio Arctic Fox (2020.3.1)

Bump protobuf gradle plugin to fix build
Also bump other unrelated dependencies and fix some linting

mafaldobot committed:
Translated using Weblate (Italian)

Currently translated at 93.3% (1110 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/it/

ashimokawa committed:
java/gradle crap: do something I do not even want to understand to fix random build errors

Yes I hate Java, and I hate gradle.
Both are a huge waste of resources.
I wonder how many nuclear plants could be switched off if people would stop using this bloated crappy VM bullshit language and "ecosystem".
And no it does not "make you super productive" and it it does not justify "investing in more hardware cpu/ram/cloud capacity" because it is "cheaper than manpower".

ashimokawa committed:
Huami: re-enable setting the timezone correctly with included DST

This will trigger bugs, but also fix some.
We "just" need to fix the new bugs :/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1189 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

gfwilliams committed:
Add functions for pushing bitmaps to Bangle.js, also add configurable MTU. If we receive a bigger data packet then we know we can send a bigger one too

Commit:
Translated using Weblate (Russian)

Currently translated at 100.0% (1189 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/ru/

ashimokawa committed:
Assume timezone UTC for BLE time protocol

BLE protocol only sends a UTC offset which combines the TZ with DST, if we take
a local timezone as a base and add a raw offset with DST included, DST might
be added again.

This is to fix problems with recorded data on Huami devices where we cannot know
in which timezone data was recorded - only the offset, where we do not know if a
part of it is DST.

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 98.5% (1172 of 1189 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

ashimokawa committed:
Amazfit T-Rex Pro: Try to fix some menu items

See #2362

vanous committed:
add Heart icon (trigger heart measurement) to device action icons

Commit:
Fossil Hybrid HR: Add timezone selector for 2nd TZ widget

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 99.9% (1191 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1192 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1192 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

Commit:
Translated using Weblate (German)

Currently translated at 100.0% (1192 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/de/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1192 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

Commit:
Fossil Hybrid HR: Add power saving feature

IhorHordiichuk committed:
Translated using Weblate (Ukrainian)

Currently translated at 100.0% (1192 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/uk/

ashimokawa committed:
Fossil Hybrid HR: Support setting metric/imperial mode

Fixes #2355

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1192 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

vanous committed:
HeartRate dialog: extend the enable check with supportsRealtimeData()

ikajdan committed:
Translated using Weblate (Polish)

Currently translated at 99.5% (1187 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

ChuckMichael committed:
Translated using Weblate (Polish)

Currently translated at 99.5% (1187 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/pl/

er2off committed:
Translated using Weblate (Russian)

Currently translated at 100.0% (1192 of 1192 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/ru/

Commit:
Translated using Weblate (Turkish)

Currently translated at 100.0% (1200 of 1200 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/tr/

IhorHordiichuk committed:
Translated using Weblate (Ukrainian)

Currently translated at 100.0% (1200 of 1200 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/uk/

chenshaoju committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (1200 of 1200 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/zh_Hans/

yarons committed:
Translated using Weblate (Hebrew)

Currently translated at 99.9% (1199 of 1200 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/he/

vincegre committed:
Translated using Weblate (French)

Currently translated at 100.0% (1200 of 1200 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/fr/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (1200 of 1200 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/nl/

vanous committed:
fix list item height when fonts are scaled up

rezaalmanda committed:
Translated using Weblate (Indonesian)

Currently translated at 18.5% (223 of 1200 strings)

Translation: Freeyourgadget/Gadgetbridge
Translate-URL: https://hosted.weblate.org/projects/freeyourgadget/gadgetbridge/id/

vanous committed:
draft changelog and other stuff for release

ashimokawa committed:
update changelogs

mjsir911 committed:
Add helpful README for inquisitive wanderers

mjsir911 committed:
Add a bit of code

mjsir911 committed:
Add uncommitted code

Commitment issues with this stuff :<
